### PR TITLE
Reporting: Track events for `View report` links on Payment activity widget

### LINKED ISSUE #8686: [Task] - Add Track events to measure click on 'View report' of each tile
To measure impact of the widget, it is necessary to add a track event to count clicks on `View report` link of each tile - Total payment volume, Deposits, Refunds, Fees and Disputes.

### PR DESCRIPTION
**Pull request by nagpai**
Fixes Automattic/woocommerce-payments#8686

#### Changes proposed in this Pull Request

Add track events to track clicks on `View report` links on the following tiles:
* Total payment volume
* Charges
* Refunds
* Disputes

#### Testing instructions

* Enable feature flag, wp option update _wcpay_feature_payment_overview_widget 1.
* Use [Tracks Vigilante](https://github.com/Automattic/tracks-vigilante-2-chrome-extension) chrome extension to track clicks from your local test site
* Check out this branch and run `npm start` or `npm run watch` to build assets
* Browse to `Payments` and on the overview page, click on the `View report` link that you see over `Total payment volume` tile. Click the link
* Check the tracked events on the Tracks Vigilante chrome extension. You should see the event `wcpay_overview_total_payment_volume_view_report_clicks` . Check the other details shared with the event, including the `source` that should read as `total_payment_volume_tile_view_report`
* Test the links on `Charges`, `Refunds` and `Disputes` tile too and see if they generate relevant track events.
###### Reminder: test this change will achieve the desired impact/result!
Reviewers please keep in mind the flow and funnel that we want to measure, and check that this event will allow us to measure the desired funnel conversion rate. i.e. test the whole flow, not just that this event fires.

- Can measure conversion rate of: View payments overview ⇒ click `View report` ⇒ view relevant report screen

See the project thread on P2 for more details and example funnel. 

#### Notes

Discussion on naming events - p1713786808704209-slack-C06BBFTF6EM

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8687` or branch name `add/8686-track-events-payment-activity-widget` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/8686-track-events-payment-activity-widget) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 83a690ef196fb12a2361da8ff2f8bb57f6667a42
- Build time: 2024-04-29 15:02:06 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/add-8686-track-events-payment-activity-widget [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: add
+
+Comment: Behind a feature flag. The PR adds track events for clicks on `View report` links on Payment activity widget
```
Comment (nagpai) [inline changelog/add-8686-track-events-payment-activity-widget]: The `npm run changelog` did not allow me to skip a change log note since thats permitted only for `type: patch` .  I have hence explicitly mentioned `Comment`

Comment (jessy-p) [reply changelog/add-8686-track-events-payment-activity-widget]: Will this work? It might cause an issue later during release when the changelogs are being processed?

File: changelog/add-8686-track-events-payment-activity-widget [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: add
+
```
Comment (shendy-a8c) [inline changelog/add-8686-track-events-payment-activity-widget:5]: I believe if you want to just add a comment that won't result a changelog entry, there shouldn't be an empty line here.

Search for `Comment:` in `changelog/` to see other 'comment' changelogs.

File: changelog/add-8686-track-events-payment-activity-widget (hunk 1 above)
Comment (nagpai) [reply changelog/add-8686-track-events-payment-activity-widget]: Thanks for catching this. @shendy-a8c  has given a nice hint on getting this fixed.

Comment (shendy-a8c) [reply changelog/add-8686-track-events-payment-activity-widget]: By the way, looking at the [GH action that generate the changelog](https://github.com/Automattic/woocommerce-payments/blob/develop/.github/workflows/release-changelog.yml), which runs this [process-changelog](https://github.com/Automattic/woocommerce-payments/blob/develop/.github/workflows/release-changelog.yml#L51) action which will run [changelogger](https://github.com/Automattic/woocommerce-payments/blob/develop/.github/actions/process-changelog/action.yml#L67), you can run this command to see what will be generated in `changelog.txt`.

This command will delete all files under `changelog/` and generate the changelog entries in `changelog.txt`:

```
./vendor/bin/changelogger write --use-version="7.6.0" --release-date="today" --no-interaction --yes
```

Comment (nagpai) [reply changelog/add-8686-track-events-payment-activity-widget]: ~Thanks! It looks like validation for changelog in CI fails if we just remove the empty line, and do not change the Significance to `patch`~

```
Error: Changelog entry may only be empty when Significance is "patch".
Error: Process completed with exit code 1.
```

Comment (nagpai) [reply changelog/add-8686-track-events-payment-activity-widget]: Phew.. fixed it :D bf77f83

**Comment by haszari** [comment]
Added note to description – reviewers please test and review these events in context, to ensure they meet the needs of project:

- Can measure conversion rate of: View payments overview ⇒ click View report ⇒ view relevant report screen

File: client/components/payment-activity/payment-activity-data.tsx [hunk 3]
```diff
@@ -83,6 +84,15 @@ const PaymentActivityData: React.FC = () => {
 					/>
 				}
 				amount={ totalPaymentVolume }
+				handleReportLinkClick={ () => {
+					recordEvent(
+						'wcpay_overview_payment_activity_total_payment_volume_click',
+						{
+							source:
+								'total_payment_volume_tile_view_report_link',
```
Comment (haszari) [inline client/components/payment-activity/payment-activity-data.tsx]: Do we need the `source` parameter (I'm assuming it's optional)? I think this is used to distinguish generic events by the originating high-level component or container. How do other WooPayments events use `source`? 

In this case we have unique events for each click target, so `source` is redundant. 

An alternative design would be to use a generic event name, and distinguish which box using source:

- en `wcpay_overview_payment_activity_click`
- source `total_payment_volume`

The question is whether we'd want to aggregate any click on the activity widget. If so, a common event is useful. If not, unique event for each thing works well.

File: client/components/payment-activity/payment-data-tile.tsx [hunk 4]
```diff
@@ -41,6 +42,10 @@ interface PaymentDataTileProps {
 	 * Optional hover link to view report.
 	 */
 	reportLink?: string;
+	/**
+	 * Optional click handler for the tile.
+	 */
+	handleReportLinkClick?: () => void;
```
Comment (haszari) [inline client/components/payment-activity/payment-data-tile.tsx]: Is our `PaymentDataTile` abstraction helping here? If we expose a prop for every detail of it, we might be better off just using `Link` directly in the outer markup. 

Alternatively, if we are ok using the same event name for all tiles e.g. `wcpay_overview_payment_tile_click`, the component could abstract away the tracking behaviour nicely. It could use an existing prop or expose a `tracksEventSource` prop, which would avoid complex callback interface.

**Review by haszari — COMMENTED** [review]
Looking good from a quick test! 

I think a generic "tile clicked" event + source prop would be cleaner and work well (and more flexible). 

I don't like passing callbacks without good reason :)

**Comment by haszari** [comment]
re-requesting a Helix review as my review was only a quick look/test.

File: client/components/payment-activity/payment-activity-data.tsx (hunk 3 above)
Comment (nagpai) [reply client/components/payment-activity/payment-activity-data.tsx]: The idea of using a common event for the tile is fantastic, and makes things totally clean! Thanks for opening my eyes to this possibility! Also a great note for me "Is this a good enough reason to pass a callback?" . Clearly no in this case with the alternative you suggested :D .  

I have implemented a common click event, and a source that is extrapolated from the existing `label` prop. ( make it all small case and replace spaces with underscores, where needed) 

I hope I have done it appropriately here : 5069046

File: client/components/payment-activity/payment-data-tile.tsx (hunk 4 above)
Comment (nagpai) [reply client/components/payment-activity/payment-data-tile.tsx]: Explained in the earlier comment. 

https://github.com/Automattic/woocommerce-payments/pull/8687#discussion_r1580961883 

Thanks!

File: client/components/payment-activity/payment-activity-data.tsx (hunk 3 above)
Comment (haszari) [reply client/components/payment-activity/payment-activity-data.tsx]: Looks good! 

I'd suggest using an explicit `source` value for each tile, rather than generating from label. This is one more prop (`tracksSource`?), but it's simple and explicit, and allows future tracks investigators to search the code for the source value to work out where it's getting fired.

Comment (nagpai) [reply client/components/payment-activity/payment-activity-data.tsx]: I have added an explicit `tracksSource` prop here - 5c5da5b . Thanks!

**Review by shendy-a8c — APPROVED** [review]
Diff looks good and tested well.
Track events were fired when I clicked on the report links from the tiles in the Overview > Payment Activity widget.
It worked on mobile as well.

By the way, don't forget to register the new track event.

**Comment by shendy-a8c** [comment]
Answering @haszari's question:

> Can measure conversion rate of: View payments overview ⇒ click View report ⇒ view relevant report screen

Is 'conversion' a click to reports from the Payment Activity widget? If so, yes, with the newly added track event from this PR, we can measure that.

If we need more than that, for example, after clicking the link, user will land on the 'report' pages.
For the TPV link, user will land on the transaction list page, unfiltered.
For the charges link, user will land on the transactions list page, filtered to only show charges.
For the refunds link, user will land on the transactions list page, filtered to only show refunds.
For the disputes link, user will land on the disputes list page.

Views to the transactions list page and disputes list page are already tracked but we can't differentiate the filtering!

Looking at the success indicator paJDYF-bWn-p2, the step 'WCPay report views' is looking at `wcadmin_page_view` with path `payments_deposits`, `payments_disputes`, and `payments_transactions`.

Screenshot 2024-04-29 at 22 31 33

Viewing the reports from the TPV link, charges link, and refunds link will lead to an increase in `wcadmin_page_view` event with path = `payments_transactions` but since the filtering isn't included in the `wcadmin_page_view` event, we will get the total number of views to the transactions list page.

I'm struggling to articulate what I mean but I hope it's clear. Otherwise, I can try to explain more.

In summary:
- If the click track event added by this PR is sufficient for us, we don't need to do anything more IMO.
- Otherwise, we need to distinguish the tracking on the transactions list page based on the filter on that page.

**Comment by haszari** [comment]
> Viewing the reports from the TPV link, charges link, and refunds link will lead to an increase in wcadmin_page_view event with path = payments_transactions but since the filtering isn't included in the wcadmin_page_view event, we will get the total number of views to the transactions list page.

I think that's fine, we want the user to click the tile and land on report/list page. As I understand it we don't particularly care what state the list view is in (from a tracking/flows POV, we do care of course that the report totals match the widget!).

We should clarify this detail with stakeholders / @souravdebnath1986 – so everyone is on the same page about how the events work.

**Comment by nagpai** [comment]
I am merging this PR based on the discussion here - p1714467303200959-slack-C06BBFTF6EM

tl;dr, the funnel tool supports adding event properties, e.g. `source`, for any step. That should help us segregate traffic by tile.

**Comment by haszari** [comment]
> By the way, don't forget to [register](https://github.com/Automattic/tracks-events-registration) the new track event.

@nagpai  checking – did you register these events as part of this or is that still needed? If not please open an issue & add to board so we remember and all our events are registered correctly.

**Comment by nagpai** [comment]
@haszari ~I have registered the event. Sorry, forgot to reference that here. Here it is!~

~https://github.com/Automattic/tracks-events-registration/pull/2396~

~What's remaining is to create the funnels. I will update here once that is done~

My apologies the PR seems to be related to an older name I was trying, before we moved to single event + different sources per tile. I will have the correct event carefully registered and update here.

**Comment by haszari** [comment]
> My apologies the PR seems to be related to an older name I was trying, before we moved to single event + different sources per tile. I will have the correct event carefully registered and update here.

Suggest just adding an issue for the outcome we need, then anyone can pick it up – best to have all tasks logged on the team board so anyone can pick up the next priority, and we have one source of truth for what's left to do. 

If there's no issue for registering all new / changed events I'd recommend adding one issue for that and mention the specific todos above so they don't get missed.

**Comment by nagpai** [comment]
>If there's no issue for registering all new / changed events I'd recommend adding one issue for that and mention the specific todos above so they don't get missed.

@haszari - Thanks for the guidance. I have added an issue to make sure the event is registered before we launch.

Automattic/tracks-events-registration#2466  - Added this to the pre-lauch check Epic too.
